### ISSUE TITLE
Multiple warnings when building develop branch

### ISSUE THREAD

**Issue by pelesh** [LLNL/hiop#267]
Several warnings pop up when building `develop` branch (without deep checks or developer mode enabled):
```shell
(...)
/hiop/src/LinAlg/hiopMatrixRajaSparseTriplet.cpp(330): warning: variable "nx" was declared but never referenced
/hiop/src/LinAlg/hiopMatrixRajaSparseTriplet.cpp(339): warning: variable "M1nnz" was declared but never referenced
/hiop/src/LinAlg/hiopMatrixRajaSparseTriplet.cpp(340): warning: variable "M2nnz" was declared but never referenced
/hiop/src/LinAlg/hiopMatrixRajaSparseTriplet.cpp(615): warning: variable "num_rows" was declared but never referenced
/hiop/src/LinAlg/hiopMatrixRajaSparseTriplet.cpp(775): warning: variable "nx" was declared but never referenced
/hiop/src/LinAlg/hiopMatrixRajaSparseTriplet.cpp(1056): warning: variable "irow_c" was declared but never referenced
/hiop/src/LinAlg/hiopMatrixRajaSparseTriplet.cpp(1058): warning: variable "irow_d" was declared but never referenced
/hiop/src/LinAlg/hiopMatrixRajaSparseTriplet.cpp(1072): warning: variable "nnz_Jac_d_new" was declared but never referenced
(...)
/hiop/src/Drivers/nlpMDS_raja_ex4.hpp(46): warning: overloaded virtual function "hiop::hiopInterfaceBase::get_starting_point" is only partially overridden in class "Ex4"
/hiop/src/Optimization/hiopNlpFormulation.hpp(460): warning: type qualifier on return type is meaningless
/hiop/src/Optimization/hiopNlpFormulation.hpp(461): warning: type qualifier on return type is meaningless
/hiop/src/Optimization/hiopNlpFormulation.hpp(462): warning: type qualifier on return type is meaningless
/hiop/src/Optimization/hiopNlpFormulation.hpp(463): warning: type qualifier on return type is meaningless
```
These seem to be minor issues, but should probably be fixed nevertheless.

**Comment by nychiang** [comment]
You gave these comments in PR#259. That one is still open, and I have addressed some of these issues in #259.
Some variables are created and used only in function `assert`. If you compile hiop in release mode, I guess these var are not used and you will receive these warnings.

**Comment by cnpetra** [comment]
closed by PR #268
